### DISCUSSION TITLE
Casting a variable as the correct type when it is read in as a string

### DISCUSSION THREAD

**Discussion by Dan-C-Reed** [gleam-lang/gleam#3100] [Questions & support] [answered]
Say I have parsed a CSV and ended up with a load of strings. Some might look like this: `"123"`. While this is a string, it should be an integer so I'd like to cast it as such. I'm looking at `gleam/dynamic` as I think this is where I'd find relevant functions. I see there are functions like `int`, `float`, etc., that check the type of a variable and returns the value if it is of that type, but in my example this would return a string, which isn't the desired result. I think I need a decoder, but I'm unclear of how to construct a decoder to convert a string to something else. How would I go about converting a string containing a `int`, `float`, or whatever into the correct data type?

Apologies for the newb question, but if someone could gently shove me in the right direction, I'd greatly appreciate it.

**Comment by jrstrunk** [comment, marked as answer]
You could use the std lib's float module! Just add `import gleam/float` then call `float.parse("123.4")`. You'll then have to handle the result type, I recommend reading that section of the tour and looking at the std lib result module on hex docs if you're not familiar :) you can use the `parse` function from the int std lib module as well

**Reply by Dan-C-Reed** [reply]
Argh — how did I miss that?! Thanks for your help, John.
